docs/src/docs/forgotPassword.adoc

```diff
@@ -37,9 +39,99 @@ The post-reset destination url is configurable in `grails-app/conf/application.g
 ----
 grails.plugin.springsecurity.ui.forgotPassword.postResetUrl = '/reset'
 ----
-
 If you don't specify a value then the `defaultTargetUrl` value will be used, which is `'/'` by default.
 
+=== Email Validation Configuration
+
+You can customize if you want to have the user get e-mails for validation before they can reset their password.  To turn off
+e-mail validation set the parameter `requireForgotPassEmailValidation` to false.  If not set it will default to true.  This
+can be set in `grails-app/conf/application.groovy` or `grails-app/conf/application.yml` under
+[source,groovy]
+----
+grails.plugin.springsecurity.ui.forgotPassword.requireForgotPassEmailValidation = false
+----
+**If e-mail validation is turned off, it is recommended to use the forgotPasswordExtraValidation below.**
+
+
+=== Challenge Questions Configuration
+
+It is recommended the provided script named `s2ui-create-challenge-questions` be used to generate challenge questions.
+
+
+
+All the answers will be encrypted and it will use the same encryption settings as the what is used to store the password
+for each user.  The service to handle this listener is automatically created and configured when
+you use the plugin.
+
+The option to add challenge questions can be turned on will work independent of e-mail validation.  If the e-mail validation is turned on this step will occur before the e-mail is sent out.
+
+To make this work, the domain object must have a link to your User Object. This is done by setting the `validationUserLookUpProperty` which defaults to 'user'.
+
+This can be customized in either `grails-app/conf/application.groovy` or `grails-app/conf/application.yml`
+[source,groovy]
+----
+grails.plugin.springsecurity.ui.forgotPassword.validationUserLookUpProperty = 'user'
+----
+
+Each list item in the `forgotPasswordExtraValidation` has three options, though each should only have two.  If you have both
+`labelDomain` and `labelMessage` then `labelDomain` will be used as it takes precedences.
+
+.forgotPasswordExtraValidation List items
+|===
+|Header 1 |Header 2
+
+|labelDomain
+|This is the property in the domain object that will used to ask the question
+
+|labelMessage
+|If labelDomain is not present then this can be used as a message property to ask a question (ie if your questions are static).
+
+|prop
+|TThis is used as the property of the domain object to determine if the answer is correct.
```

Comment:
Apologies, but I found one more small typo... `TThis` should be `This`

Reply:
Fixed in next commit